packages/cardano-services/src/util/TypeormProvider/TypeormProvider.ts

```diff
@@ -1,34 +1,31 @@
-import { BehaviorSubject, Observable, Subscription, filter, firstValueFrom, tap } from 'rxjs';
+import { BehaviorSubject, Observable, tap } from 'rxjs';
 import { DataSource } from 'typeorm';
 import { HealthCheckResponse, Provider } from '@cardano-sdk/core';
 import { Logger } from 'ts-log';
 import { PgConnectionConfig } from '@cardano-sdk/projection-typeorm';
-import { RunnableModule, isNotNil } from '@cardano-sdk/util';
-import { createTypeormDataSource } from './util';
+import { TypeormService } from '../TypeormService';
+import { createTypeormDataSource } from '../createTypeormDataSource';
 
 export interface TypeormProviderDependencies {
   logger: Logger;
   entities: Function[];
   connectionConfig$: Observable<PgConnectionConfig>;
 }
 
-export abstract class TypeormProvider extends RunnableModule implements Provider {
+export abstract class TypeormProvider extends TypeormService implements Provider {
   #entities: Function[];
-  #subscription: Subscription | undefined;
   #connectionConfig$: Observable<PgConnectionConfig>;
   #dataSource$ = new BehaviorSubject<DataSource | null>(null);
-
-  logger: Logger;
   health: HealthCheckResponse = { ok: false, reason: 'not started' };
 
   constructor(name: string, { connectionConfig$, logger, entities }: TypeormProviderDependencies) {
-    super(name, logger);
+    super(name, { connectionConfig$, entities, logger });
     this.#entities = entities;
     this.#connectionConfig$ = connectionConfig$;
   }
 
   #subscribeToDataSource() {
```

Comment:
We should keep source subscription logic in base class and instead of re-subscribing to source, subscribe to the subject to track health.
```suggestion
    this.#subscription = this.#dataSource.subscribe(() => (this.health = { ok: true }))
  }
```

Reply:
Doing this breaks [this](https://github.com/input-output-hk/cardano-js-sdk/blob/0db5cb6c077cefdf1422270300a7416090166c59/packages/cardano-services/test/StakePool/TypeormStakePoolProvider/TypeormStakePoolProvider.test.ts#L145-L154) test.

Reply:
Please push a commit that does this and let's try to figure out what's happening

Reply:
If you look carefully at the suggestion it removes `#subscribeToSource` and subscribes to Subject in the constructor

Reply:
So sorry, I missed that. Working on that